sled-agent/src/bootstrap/measurements.rs

```diff
@@ -0,0 +1,67 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Functions related to management of measurement corpus
+use camino::{Utf8Path, Utf8PathBuf};
+//use sled_storage::dataset::INSTALL_DATASET;
+//use sled_storage::manager::StorageHandle;
+use sled_agent_config_reconciler::InternalDisksReceiver;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum MeasurementError {
+    #[error("Missing INSTALL dataset")]
+    MissingInstallSet,
+    #[error("io: {0}")]
+    Io(std::io::Error),
+    #[error("Missing boot disk")]
+    MissingBootDisk,
+}
+
+/// Access the measurements in the install directory
+pub async fn sled_new_measurement_paths(
+    receiver: &InternalDisksReceiver,
+) -> Result<Vec<Utf8PathBuf>, MeasurementError> {
+    let mut all = vec![];
+    let mut dirs = vec![];
+
+    let current = receiver.current();
+
+    dirs.push(
+        current
+            .boot_disk_install_dataset()
+            .ok_or(MeasurementError::MissingBootDisk)?
+            .join("measurements"),
```

Comment:
@jgallagher this the recommended way to get the install dataset? I _think_ by the time we're calling these functions the boot disks should have come up. I also wasn't sure if we should just be doing something like
```
    let paths: Vec<_> = internal_disks_rx
        .current()
        .all_install_datasets()
        .map(|p| p.join("measurements"))
        .collect();
```
with a new API